### PR TITLE
FIX: Minimize scope for directory changes while loading results file

### PR DESCRIPTION
**Pull request by effigies**
## Summary
This PR attempts to handle the results file failures by separating concerns.

1) Load the results file without changing directories. We are pretty consistently seeing "No file named `results_<whatever>.pklz`, which at the minimum is uninformative. There's no reason not to know the full path of the file when we try to open it.

2) Perform `pickle.loads` on the file contents as a single-line `with indirectory` context block.

3) All other operations are directory independent and file-lock independent.

Alternative to #3047.

Sorry to preempt that process, but we're seeing this a lot, and I don't really know how to help move that one along. I'm going to use this branch to see if we can resolve these issues that are popping up a lot in the fMRIPrep tests.

## Acknowledgment

- [x] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/utils/filemanip.py [hunk 1]
```diff
@@ -699,37 +699,49 @@ def loadpkl(infile):
     infile = Path(infile)
     fmlogger.debug('Loading pkl: %s', infile)
     pklopen = gzip.open if infile.suffix == '.pklz' else open
+
+    with SoftFileLock('%s.lock' % infile.name):
+        with pklopen(str(infile), 'rb') as pkl_file:
+            pkl_contents = pkl_file.read()
```
Comment (oesteban) [inline nipype/utils/filemanip.py]:
```suggestion
        pkl_contents = infile.read_bytes()
```
although we can wait until only Python +3.5 is supported.

Comment (effigies) [reply nipype/utils/filemanip.py]: Yeah, I feel like we've been bitten by trying to jump ahead of ourselves. I'd rather stick with idioms universal to our support window.

Comment (effigies) [reply nipype/utils/filemanip.py]: Oh, and actually that wouldn't work, since we need to use `gzip.open` for `pklz` files anyway.

**Comment by effigies** [comment]
dcm2niix bug cropping up again. Merging master. Please see https://github.com/poldracklab/fmriprep/pull/1800 and https://github.com/poldracklab/fmriprep/pull/1803 for fixed tests.

**Pull request merged**